graphql/schema/directives.graphql

```diff
@@ -17,6 +22,31 @@ directive @requireProjectSettingsAccess on FIELD_DEFINITION
 requireDistroAccess is used to restrict view, edit, admin, and create access for distros.
 """
 directive @requireDistroAccess(access: DistroSettingsAccess!) on ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION
+enum DistroSettingsAccess {
+  ADMIN
+  CREATE
+  EDIT
+  VIEW
+}
+
+"""
+requireTaskAccess is used to restrict view, edit, and admin access for tasks.
+"""
+directive @requireTaskAccess(access: TaskAccess!) on ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION
```

Comment:
I know you mentioned merging project directives in the PR description, but was wondering if you considered simply having one general access resolver? Something like this:

```suggestion
directive @requireAccess(resource: Resource!, access: AccessLevel!) on ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION

enum Resource {
  ANNOTATION
  TASK
}

enum AccessLevel {
  ADMIN
  CREATE
  EDIT
  VIEW
}
```

I feel like the code in resolver.go has a lot of repetition and maybe this could help us reduce that, and make it easier to scale resolvers across the codebase. But maybe there is a limitation to this approach I'm not thinking of. Interested in anyone's thought's too since this is going to be used a lot in the future!

Reply:
I think it might be possible if the [resource](https://github.com/evergreen-ci/evergreen/blob/5faf7d17fb1a7553202dfcf0e0ef94aa133afa54/globals.go#L1124-L1126) is one of `Superuser | Project | Distro`, the permissions are one of [these options](https://github.com/evergreen-ci/evergreen/blob/5faf7d17fb1a7553202dfcf0e0ef94aa133afa54/globals.go#L1145-L1160), and the access levels are similar to what you wrote

I'm not sure if I would want to have one huge directive though, it could be a bit difficult to maintain. I think a big Project directive, Distro directive, and Superuser directive could be make sense

Having one directive for the Project resource type would be a matter of calling the function `GetProjectIdForProjectScopes` to get the project ID depending on whatever parameters are provided in the resolver's input, and something similar to the input parameters you wrote, but what do you think haha